src/pykeen/models/base.py

```diff
@@ -202,29 +217,42 @@ def _new_init(self, *args, **kwargs):
         _original_init(self, *args, **kwargs)
         self.reset_parameters_()
 
-    cls.__init__ = _new_init
+    # sorry mypy, but this kind of evil must be permitted.
+    cls.__init__ = _new_init  # type: ignore
 
 
 class Model(nn.Module, ABC):
-    """A base module for all of the KGE models."""
+    """An abstract class for knowledge graph embedding models (KGEMs).
+
+    The only function that needs to be implemented for a given subclass is
+    :meth:`Model.forward`. The job of the :meth:`Model.forward` function, as
+    opposed to the completely general :meth:`torch.nn.Module.forward` is
+    to take indices for the head, relation, and tails' respective representation(s)
+    and to determine a score.
+
+    Subclasses of Model can decide however they want on how to store entities' and
+    relations' representations, how they want to be looked up, and how they should
+    be scored. The :class:`ERModel` provides a commonly useful implementation
+    which allows for the specification of one or more entity representations and
+    one or more relation representations in the form of :class:`pykeen.nn.Embedding`
+    as well as a matching instance of a :class:`pykeen.nn.Interaction`.
+    """
 
     #: A dictionary of hyper-parameters to the models that use them
     _hyperparameter_usage: ClassVar[Dict[str, Set[str]]] = defaultdict(set)
 
+    #: Keep track of if this is a base model
+    _is_abstract: ClassVar[bool]
+
     #: The default strategy for optimizing the model's hyper-parameters
     hpo_default: ClassVar[Mapping[str, Any]]
+
     #: The default loss function class
     loss_default: ClassVar[Type[Loss]] = MarginRankingLoss
     #: The default parameters for the default loss function class
     loss_default_kwargs: ClassVar[Optional[Mapping[str, Any]]] = dict(margin=1.0, reduction='mean')
     #: The instance of the loss
     loss: Loss
-    #: The default regularizer class
-    regularizer_default: ClassVar[Type[Regularizer]] = NoRegularizer
-    #: The default parameters for the default regularizer class
-    regularizer_default_kwargs: ClassVar[Optional[Mapping[str, Any]]] = None
-    #: The instance of the regularizer
-    regularizer: Regularizer
```

Comment:
if it's not saved in the model, where should default regularization information be saved?